src/core0_audio.rs

```diff
@@ -0,0 +1,676 @@
+                    synced_date_time.get_timestamp_micros(&timer),
+                ),
+                &mut flash_storage,
+            );
+            logged_power_down = true;
+        }
+        watchdog.feed();
+        if should_sleep {
+            if let Ok(pi_is_powered_down) = shared_i2c.pi_is_powered_down(&mut delay, true) {
+                if pi_is_powered_down {
+                    if alarm_time.is_some() {
+                        let until_alarm = (alarm_time.unwrap()
+                            - synced_date_time.get_adjusted_dt(timer))
+                        .num_minutes();
+
+                        if until_alarm < 1 {
+                            // otherwise the alarm could trigger  between here and sleeping
+                            should_sleep = false;
+                            info!("Alarm is scheduled in {} so not sleeping", until_alarm);
+                            if until_alarm <= 0 {
+                                watchdog.start(100.micros());
+                                loop {
+                                    // Wait to be reset and become thermal device
+                                    nop();
+                                }
+                            }
+                            continue;
+                        }
+                    }
+                    info!("Ask Attiny to power down rp2040");
+                    event_logger.log_event(
+                        LoggerEvent::new(
+                            LoggerEventKind::Rp2040Sleep,
+                            synced_date_time.get_timestamp_micros(&timer),
+                        ),
+                        &mut flash_storage,
+                    );
+
+                    if let Ok(_) = shared_i2c.tell_attiny_to_power_down_rp2040(&mut delay) {
+                        info!("Sleeping");
+                    } else {
+                        error!("Failed sending sleep request to attiny");
+                    }
+                }
+
+                //may aswell offload again if we are awake and have just taken a recording
+                if !pi_is_powered_down && flash_storage.has_files_to_offload() {
+                    match offload(
+                        &mut shared_i2c,
+                        clock_freq,
+                        &mut flash_storage,
+                        &mut pi_spi,
+                        timer,
+                        &mut event_logger,
+                        should_wake,
+                        device_config,
+                        &mut delay,
+                        &synced_date_time,
+                        watchdog,
+                    ) {
+                        Ok((new_config, was_updated)) => {
+                            device_config = new_config.unwrap();
+                            if was_updated {
+                                if !device_config.config().is_audio_device {
+                                    shared_i2c.disable_alarm(&mut delay);
```

Comment:
Handle `Result`?

Reply:
this gets disbaled in main anyway so will remove it